integration/spark/shared/src/main/java/io/openlineage/spark/api/OpenLineageContext.java

```diff
@@ -145,4 +146,11 @@ public String getSparkVersion() {
    * file. All the configuration should be read from this object.
    */
   @NonNull @Getter final SparkOpenLineageConfig openLineageConfig;
+
+  /**
+   * Helper class that uses reflection to call methods of SparkOpenLineageExtensionVisitor
+   * instances. This class acts as a bridge, facilitating communication between openlineage-spark
+   * and the extension implementations provided by spark-interfaces-shaded.
+   */
+  @NonNull @Getter final SparkOpenLineageExtensionVisitorWrapper sparkExtensionVisitorWrapper;
```

Comment:
Why is this "configurable" per context rather than single class for whole integration? Is there use case where there can be different `SparkOpenLineageExtensionVisitorWrapper`?